src/main/java/it/pagopa/pn/delivery/middleware/notificationdao/EntityToDtoMapper.java

```diff
@@ -0,0 +1,153 @@
+package it.pagopa.pn.delivery.middleware.notificationdao;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectReader;
+import it.pagopa.pn.api.dto.notification.*;
+import it.pagopa.pn.delivery.model.notification.cassandra.NotificationEntity;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+//@Mapper( componentModel = "spring", unmappedTargetPolicy = ReportingPolicy.ERROR)
+@Component
+public class EntityToDtoMapper {
+
+    private final ObjectMapper objMapper;
+    private final ObjectReader recipientReader;
+
+    public EntityToDtoMapper(ObjectMapper objMapper) {
+        this.objMapper = objMapper;
+        this.recipientReader = this.objMapper.readerFor( NotificationRecipient.class );
+    }
+
+    public Notification entity2Dto(NotificationEntity entity) {
+        Notification.NotificationBuilder builder = Notification.builder()
+                .iun( entity.getIun() )
+                .subject( entity.getSubject() )
+                .paNotificationId( entity.getPaNotificationId() )
+                .cancelledByIun( entity.getCancelledByIun() )
+                .cancelledIun( entity.getCancelledIun() )
+
+                .sender( NotificationSender.builder()
+                        .paId( entity.getSenderPaId() )
+                        .build()
+                    )
+                .recipients( buildRecipientsList( entity ) )
+
+                .documents( buildDocumentsList( entity ) );
+
+        boolean anyPaymentFieldNotNull = Arrays.asList(
+                    entity.getIuv(), entity.getNotificationFeePolicy(),
+                    entity.getF24AnalogDigestSha256(), entity.getF24AnalogVersionId(),
+                entity.getF24DigitalDigestSha256(), entity.getF24DigitalVersionId(),
+                entity.getF24FlatRateDigestSha256(), entity.getF24FlatRateVersionId()
+                )
+                .stream().anyMatch( val -> val != null);
+        if ( anyPaymentFieldNotNull ) {
+            builder.payment( buildNotificationPaymentInfo( entity ) );
+        }
+
+        // Generated from timeline table
+        //.notificationStatus()
+        //.notificationStatusHistory()
+        //.timeline()
+
+        return builder.build();
+    }
+
+    private NotificationPaymentInfo buildNotificationPaymentInfo(NotificationEntity entity) {
```

Comment:
**Manutenibilità**

Non è bloccante, ma estrarrei in altre classi le trasformazioni di sotto-parti de modello, se sono "complesse".
C'è molta logica in questa classe ora, e questo ha effetto sui test che devono andare a provare molte + combinazioni di quando si farebbe altrimenti.

Lo stesso vale per gli attachment, ma non ripeto il commento.

Consiglio di disaccoppiare in sostanza.